src/tests/integration/test_prove.py

```diff
@@ -159,9 +161,41 @@ def test_prove_equivalence(
         max_iterations=MAX_ITERATIONS,
     )
 
-    proof_show = symtool.proof_show
-    show_result = '\n'.join(proof_show.show(proof, [node.id for node in proof.kcfg.nodes]))
-    (symtool.proof_dir / f'{test_id.upper()}-proof-result.txt').write_text(show_result)
-
     # Then: Prove `R(S_{REVM}.initial, S_{KEVM}.initial) /\ R(S_{REVM}.final, S_{KEVM}.final)`
     # `R` is the relation between KEVM state `S_{KEVM}` and REVM State in RISC-V memory `S_{REVM}`
+    report = check_proof(proof, symtool)
+    report += '\n'.join(symtool.proof_show.show(proof, [node.id for node in proof.kcfg.nodes]))
+    (symtool.proof_dir / f'{test_id.upper()}-proof-report.txt').write_text(report)
+
+
+def collect_int2bytes(term: KInner) -> list[KInner]:
+    """Collect all `Int2Bytes` in the term."""
+    from pyk.kast.inner import KLabel, collect
+
+    int2bytes_list: list[KInner] = []
+
+    def _collect_int2bytes(kinner: KInner) -> None:
+        match kinner:
+            case KApply(KLabel('Int2Bytes(_,_,_)_BYTES-HOOKED_Bytes_Int_Int_Endianness')):
+                int2bytes_list.append(kinner)
+
+    collect(_collect_int2bytes, term)
+    return int2bytes_list
+
+
+def check_proof(proof: APRProof, symtool: SymTools) -> str:
+    from kriscv.utils import kast_print
+
+    report: list[str] = []
+
+    for node in proof.kcfg.nodes:
+        # forall cterm, there is no `Int2Bytes` in their `regs` cells
+        regs = node.cterm.cell('REGS_CELL')
+        int2bytes_list = collect_int2bytes(regs)
+        while int2bytes_list:
+            int2bytes = int2bytes_list.pop()
```

Comment:
```suggestion
        for int2bytes in int2bytes_list:
```

(No need to empty the list, the garbage collector will eventually take care of it once it goes out of scope.)